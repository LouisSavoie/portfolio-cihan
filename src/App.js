import './App.scss'
import React, { useState, useEffect } from 'react'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import Navbar from './components/Navbar/Navbar'
import Info from './components/Info/Info'
import Projects from './components/Projects/Projects'
import Blog from './components/Blog/Blog'

function App () {
  const [data, setData] = useState([])

  const getData = () => {
    fetch('info.json',
      {
        headers: {
          'Content-Type': 'application/json',
          Accept: 'application/json'
        }
      })
      .then(function (response) { return response.json() })
      .then(function (responseJson) { setData(responseJson) })
  }

  useEffect(() => {
    getData()
  }, [])

  return (
    <div data-testid="app">
      <Router>
        <Navbar name={data.name} />
        <Routes>
          <Route path="/" element={<Info data={data} />} />
          <Route path="/projects" element={<Projects projects={data.projects} />} />
          <Route path="/blog" element={<Blog blogName={data.blog} />} />
        </Routes>
      </Router>
    </div>
  )
}

export default App
