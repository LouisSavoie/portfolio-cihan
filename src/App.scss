@import "~bootstrap/scss/bootstrap";

// COLORS

// Light Theme (Green)
$nav: rgb(26, 60, 64); // Dark
$nav-link: rgb(237, 230, 219); // Light
$nav-link-alt: white; // White
$heading: rgb(65, 125, 122); // Medium
$text: rgb(26, 60, 64); // Dark
$button: rgb(29, 92, 99); // Darker Medium
$button-alt: rgb(65, 125, 122); // Medium
$button-text: rgb(237, 230, 219); // Light
$card-bg: rgba(255, 255, 255, 0.45); // 45% White
$bg: rgb(237, 230, 219); // Light

// Light Theme (Blue)
// $nav: rgb(54, 48, 98); // Dark
// $nav-link: rgb(233, 213, 218); // Light
// $nav-link-alt: white; // White
// $heading: rgb(130, 115, 151); // Medium
// $text: rgb(54, 48, 98); // Dark
// $button: rgb(77, 76, 125); // Darker Medium
// $button-alt: rgb(130, 115, 151); // Medium
// $button-text: rgb(233, 213, 218); // Light
// $card-bg: rgba(255, 255, 255, 0.45); // 45% White
// $bg: rgb(233, 213, 218); // Light

// Dark Theme (Green)
// $nav: rgb(29, 92, 99); // Darker Medium
// $nav-link: rgb(237, 230, 219); // Light
// $nav-link-alt: white; // White
// $heading: rgb(237, 230, 219); // Light
// $text: rgb(65, 125, 122); // Medium
// $button: rgb(29, 92, 99); // Darker Medium
// $button-alt: rgb(65, 125, 122); // Medium
// $button-text: rgb(237, 230, 219); // Light
// $card-bg: rgba(0, 0, 0, 0.30); // 30% Black
// $bg: rgb(26, 60, 64); // Dark

// Dark Theme (Blue)
// $nav: rgb(77, 76, 125); // Darker Medium
// $nav-link: rgb(233, 213, 218); // Light
// $nav-link-alt: white; // White
// $heading: rgb(233, 213, 218); // Light
// $text: rgb(130, 115, 151); // Medium
// $button: rgb(77, 76, 125); // Darker Medium
// $button-alt: rgb(130, 115, 151); // Medium
// $button-text: rgb(233, 213, 218); // Light
// $card-bg: rgba(0, 0, 0, 0.30); // 30% Black
// $bg: rgb(54, 48, 98); // Dark

// Elements

body {
  background-color: $bg;
}

h1, h3, h4 {
  color: $heading;
}

h2 {
  padding-bottom: 1.5rem;
  color: $bg;
}

a, a:hover, a:focus {
  text-decoration: none;
}

// Classes

.card-img-top-center-cropped {
  object-fit: cover;
  object-position: center;
  height: 236px;
  width: 100%;
}

// Bootstrap Overrides

.navbar-custom {
  background-color: $nav;
}

.navbar-brand {
  color: $nav-link;
}

.nav-link, .nav-link:visited {
  color: $nav-link;
}

.nav-link:hover, .nav-link:active, .nav-link:focus {
  color: $nav-link-alt;
}

.navbar-toggler svg {
  color: $nav-link;
}

.btn {
  background-color: $button;
  border-color: $button;
  color: $button-text;
  border-radius: 50rem;
}

.btn:hover, .btn:focus {
  background-color: $button-alt;
  border-color: $button-alt;
  color: $button-text;
}

.card-img-top {
  border: 1px;
  border-radius: 1rem 1rem 0 0;
}

.card {
  margin: 1rem;
  padding: 0;
  background-color: $card-bg;
  border: 1px;
  border-radius: 1rem;
}

.card-body {
  background-color: rgb(0,0,0,0);
  border: 1px;
  border-radius: 0 0 1rem 1rem;
}

.card-text, .card-link {
  color: $text;
}
