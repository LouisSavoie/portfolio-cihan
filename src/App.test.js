import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import App from './App'

afterEach(() => {
  cleanup()
})

test('should render App', () => {
  render(<App />)
  const AppElement = screen.getByTestId('app')
  expect(AppElement).toBeInTheDocument()
})

test('matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
