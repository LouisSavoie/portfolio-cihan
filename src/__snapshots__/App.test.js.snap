// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="app"
>
  <nav
    className="navbar navbar-expand-sm navbar-custom fixed-top"
    data-testid="navbar"
  >
    <div
      className="container"
    >
      <h1
        className="navbar-brand"
      />
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bars "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
        />
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="active nav-link"
            data-rr-ui-event-key="/"
            disabled={false}
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Info
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/projects"
            disabled={false}
            href="/projects"
            onClick={[Function]}
            style={null}
          >
            Projects
          </a>
          <a
            className="nav-link"
            data-rr-ui-event-key="/blog"
            disabled={false}
            href="/blog"
            onClick={[Function]}
            style={null}
          >
            Blog
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    data-testid="info"
  >
    <h2>
      Info
    </h2>
    <div
      className="container"
      data-testid="info-panel"
    >
      <div
        className="card"
      >
        <div
          className="justify-content-center row row-cols-md-2 row-cols-sm-1 row-cols-1"
        >
          <div
            className="pe-md-0 col-xxl-4 col"
          >
            <img
              className="card-img"
              style={
                Object {
                  "borderRadius": "1rem",
                }
              }
            />
            <div
              className="d-none d-md-flex d-lg-none justify-content-center flex-wrap card-body"
            >
              <a
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                rel="noopener noreferrer"
                role="button"
                tabIndex={0}
                target="_blank"
              >
                Resume
              </a>
              <a
                className="btn btn-primary"
                disabled={false}
                href="mailto:undefined"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                Email
              </a>
              <a
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                rel="noopener noreferrer"
                role="button"
                tabIndex={0}
                target="_blank"
              >
                LinkedIn
              </a>
              <a
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                rel="noopener noreferrer"
                role="button"
                tabIndex={0}
                target="_blank"
              >
                GitHub
              </a>
              <a
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                rel="noopener noreferrer"
                role="button"
                tabIndex={0}
                target="_blank"
              >
                LeetCode
              </a>
              <a
                className="btn btn-primary"
                disabled={false}
                onClick={[Function]}
                onKeyDown={[Function]}
                rel="noopener noreferrer"
                role="button"
                tabIndex={0}
                target="_blank"
              >
                Codewars
              </a>
            </div>
          </div>
          <div
            className="ps-md-0 col-xxl-8 col"
          >
            <div
              className="card-body"
            >
              <h1
                className="text-center card-title"
              />
              <h3
                className="text-center card-title"
              />
              <div
                className="d-xs-flex d-sm-flex d-md-none d-lg-flex justify-content-center flex-wrap"
              >
                <a
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  rel="noopener noreferrer"
                  role="button"
                  tabIndex={0}
                  target="_blank"
                >
                  Resume
                </a>
                <a
                  className="btn btn-primary"
                  disabled={false}
                  href="mailto:undefined"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Email
                </a>
                <a
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  rel="noopener noreferrer"
                  role="button"
                  tabIndex={0}
                  target="_blank"
                >
                  LinkedIn
                </a>
                <a
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  rel="noopener noreferrer"
                  role="button"
                  tabIndex={0}
                  target="_blank"
                >
                  GitHub
                </a>
                <a
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  rel="noopener noreferrer"
                  role="button"
                  tabIndex={0}
                  target="_blank"
                >
                  LeetCode
                </a>
                <a
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  rel="noopener noreferrer"
                  role="button"
                  tabIndex={0}
                  target="_blank"
                >
                  Codewars
                </a>
              </div>
              <p
                className="card-text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
